src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -186,6 +195,13 @@ private unsafe QuicConnection()
                 &handle),
                 "ConnectionOpen failed");
             _handle = new MsQuicContextSafeHandle(handle, context, SafeHandleType.Connection);
+
+#if DEBUG
+            if (KeyLogFile != null)
+            {
+                _tlsSecret = new MsQuicTlsSecret(_handle);
+            }
+#endif
```

Comment:
This should be outside of try catch to correspond to  the logic of inbound connection, or vice versa.